dace/frontend/python/newast.py

```diff
@@ -3178,6 +3178,12 @@ def visit_AnnAssign(self, node: ast.AnnAssign):
         try:
             dtype = eval(astutils.unparse(node.annotation), self.globals,
                          self.defined)
+            if isinstance(dtype, data.Data):
+                simple_type = dtype.dtype
+            else:
+                simple_type = dtype
+            if not isinstance(simple_type, dtypes.typeclass):
+                raise
```

Comment:
@alexnick83 raise what? should this be `TypeError('Type annotations of non-dace types not supported')` ?

Reply:
it is raising so that it is caught by the `except` below.

Reply:
We could make it NotImplementedError or TypeError but I am not about the policy. It appears that the idea now is to just warn and ignore the annotation.

Reply:
The added test reflects that.